WORKSPACE

```diff
@@ -15,3 +15,27 @@ envoy_dependencies()
 load("//bazel:dependency_imports.bzl", "envoy_dependency_imports")
 
 envoy_dependency_imports()
+
+load("@rules_python//python:repositories.bzl", "py_repositories")
+
+py_repositories()
+
+load("@rules_python//python:pip.bzl", "pip_repositories")
+
+pip_repositories()
+
+load("@rules_python//python:pip.bzl", "pip3_import")
+
+# Create a central repo that knows about the dependencies needed for
+# requirements.txt.
+pip3_import(
+    # or pip3_import
+    name = "protodoc_deps",
+    requirements = "//tools/protodoc:requirements.txt",
+)
+
+# Load the central repo's install function from its `//:requirements.bzl` file,
+# and call it.
+load("@protodoc_deps//:requirements.bzl", "pip_install")
```

Comment:
@lizan I'm trying to roll this into `envoy_dependency_imports()` and `envoy_dependencies()`. Unfortunately, I think this requires a 3 stage `load(..)`/invoke function from load sequence. The core issue is that we can't do the `pip3_import` in `envoy_dependencies()`, since it needs `@rules_protobuf`, which itself is only loaded from `repository_locations.bzl` in `repositories.bzl`.

Any ideas on how to do this in 2 stages? If not, I think we need to add a `envoy_dependencies_extra()` stage to this flow.

Reply:
https://github.com/envoyproxy/envoy/pull/11108